### PR TITLE
[PyOV] Fix deprecation warnings for gcc

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - *item1*
 - *...*

### Tickets:
 - CVS-139736
 - CVS-139194

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/utils/utils.hpp [hunk 1]
```diff
@@ -12,10 +12,15 @@
     // Generated by operator `==` inside pybind::handle.
     #pragma warning( push )
     #pragma warning( disable: 4996 )
+#elif defined(__GNUC__)
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/pyopenvino/utils/utils.hpp:17]: looks like deprecated API ov::Any is used, can we avoid it?

Comment (akuporos) [reply src/bindings/python/src/pyopenvino/utils/utils.hpp:17]: No, there is another problem:
`/home/akupuros/openvino/openvino/src/core/include/openvino/core/any.hpp:449:20: warning: ‘bool pybind11::handle::operator==(const pybind11::handle&) const’ is deprecated: Use obj1.is(obj2) instead [-Wdeprecated-declarations]
  449 |         return rhs == lhs;`
  
Warning occurred at the junction of pybind11 and the templates inside ov::Any.

File: src/bindings/python/src/pyopenvino/frontend/manager.cpp [hunk 2]
```diff
@@ -99,68 +99,68 @@ void regclass_frontend_FrontEndManager(py::module m) {
...
-    static py::exception<ov::frontend::OpValidationFailure> exc(std::move(m), "OpValidationFailure");
+    static py::handle ex =
+        py::exception<ov::frontend::OpValidationFailure>(std::move(m), "OpValidationFailure").release();
     py::register_exception_translator([](std::exception_ptr p) {
         try {
             if (p)
                 std::rethrow_exception(p);
         } catch (const ov::frontend::OpValidationFailure& e) {
-            exc(e.what());
+            py::set_error(ex, e.what());
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/pyopenvino/frontend/manager.cpp:124]: it's not available in older versions of pybind https://openvino-ci.toolbox.iotg.sclab.intel.com/blue/organizations/jenkins/private-ci%2Fie%2Fbuild-linux-macos_arm64/detail/build-linux-macos_arm64/15687/pipeline#step-402-log-99

as it's stated in tickets, we need to keep compatibility.

Comment (akuporos) [reply src/bindings/python/src/pyopenvino/frontend/manager.cpp:122]: done

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**